hre/src/main/java/hre/lang/System.java

```diff
@@ -158,11 +161,22 @@ private static void log(LogLevel level, Map<Appendable, LogLevel> outputs, Strin
                         for (LogWriter writer : activeLogWriters) {
                             writer.doFlush();
                         }
+                        message = level.getShorthand() + String.format(format, args);
+                    }
 
-                        message = level.getShorthand() + String.format(format + "%n", args);
+                    if(needLineClear /* && Config.ansi true below */) {
+                        entry.getKey().append("\033[0K");
+                        needLineClear = false;
                     }
 
                     entry.getKey().append(message);
+
+                    if(level == LogLevel.Progress && Configuration.ansi.get()) {
+                        entry.getKey().append("\r");
+                        needLineClear = true;
+                    } else {
+                        entry.getKey().append("\r\n");
```

Comment:
No, the point is to do a true carriage return (and no newline), and then clear the line. I don't know if it's acceptable to do only "\n" instead of "\r\n" in this case, but this works, so...

Reply:
If you want to write new cross-platform code, you should use a format string with "%n" by the way.